src/MusicalScore/VoiceData/ChordSymbolContainer.ts

```diff
@@ -45,35 +52,94 @@ export class ChordSymbolContainer {
                 transposeHalftones
             );
         }
+        if (chordSymbol.ChordKind === ChordSymbolEnum.none) {
+            return chordSymbol.getTextFromChordKindEnum(chordSymbol.ChordKind);
+        }
         // main Note
         let text: string = Pitch.getNoteEnumString(transposedRootPitch.FundamentalNote);
         // main alteration
         if (transposedRootPitch.Accidental !== AccidentalEnum.NONE) {
             text += this.getTextForAccidental(transposedRootPitch.Accidental);
         }
+
+        // degrees
+        const degrees: DegreesInfo = {
+            adds: [],
+            alts: [],
+            subs: [],
+        };
+
+        for (const chordDegree of chordSymbol.ChordDegrees) {
+            if (chordDegree) {
+                let t: string = "";
+                if (chordDegree.alteration !== AccidentalEnum.NONE) {
+                    t += this.getTextForAccidental(chordDegree.alteration);
+                }
+                t += chordDegree.value;
+                switch (chordDegree.text) {
+                    case ChordDegreeText.add:
+                        degrees.adds.push(t);
+                        break;
+                    case ChordDegreeText.alter:
+                        degrees.alts.push(t);
+                        break;
+                    case ChordDegreeText.subtract:
+                        degrees.subs.push(t);
+                        break;
+                    default:
+                }
+            }
+        }
+
         // chord kind text
-        text += chordSymbol.getTextFromChordKindEnum(chordSymbol.ChordKind);
-        // degree
-        if (chordSymbol.ChordDegree) {
-            switch (chordSymbol.ChordDegree.text) {
-                case ChordDegreeText.add:
-                    text += "add";
-                    text += chordSymbol.ChordDegree.value.toString();
-                    break;
-                case ChordDegreeText.alter:
-                    if (chordSymbol.ChordDegree.alteration !== AccidentalEnum.NONE) {
-                        text += this.getTextForAccidental(chordSymbol.ChordDegree.alteration);
+        // I'm going to store this in a variable for now so I can evaluate it with the degrees
+        let chordKind: string = chordSymbol.getTextFromChordKindEnum(chordSymbol.ChordKind);
+        const degreeTypeAry: string[] = ["adds", "alts", "subs"];
+
+        const customChordKinds: CustomChordKind[] = chordSymbol.rules.CustomChordKinds;
+
+        for (const customKind of customChordKinds) {
+            if (
+                customKind.chordKind !== chordSymbol.chordKind
+            ) {
+                continue;
+            }
+
+            let check: boolean = true;
```

Comment:
this variable shouldn't be named so generically.
Judging from the following code, something like `hasCustomKindDegrees` could be the right direction.